src/events/mqtt.js

```diff
@@ -97,10 +162,58 @@ function subscribeWrapper(originalSubscribeFunc) {
             tracer.addException(err);
         }
 
-        return originalSubscribeFunc.apply(this, [topic, options, patchedCallback]);
+        return originalSubscribeFunc.apply(this, [topic, internalOptions, patchedCallback]);
     };
 }
 
+/**
+ * Wraps the on command function with tracing
+ * @param {Function} originalOnFunc The wrapped function
+ * from mqtt module
+ * @returns {Function} The wrapped function
+ */
+function onWrapper(originalOnFunc) {
+    return function internalOnWrapper(eventName, callback) {
+        let patchedCallback = callback;
+        try {
+            if (eventName === 'message') {
+                const responseMetadata = {
+                    region: this.options.region,
+                    protocol: this.options.protocol,
+                    host: this.options.host,
+                };
+                const payload = {
+                    clientId: this.options.clientId,
+                    protocolId: this.options.protocolId,
+                    protocolVersion: this.options.protocolVersion,
+                };
+                patchedCallback = (topic, message, ...rest) => {
+                    const { slsEvent: mqttEvent, startTime } = eventInterface.initializeEvent('mqtt', topic, 'onMessage', 'mqtt');
+                    payload.message = message ? message.toString() : message;
```

Comment:
are you sure about the `toString()` here (forgot to talk to you about it)? won't we get `[Object object]`? or is it never an object and always string / bytes?

Reply:
I saw that it always bytes